src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -2374,11 +2629,6 @@ private void EnterScope(WriterState newState, ODataItem item)
                                     nestedResourceInfo.IsCollection = navigationProperty.Type.IsEntityCollectionType();
                                 }
 
-                                if (!nestedResourceInfo.IsCollection.HasValue)
-                                {

```

Comment:
Why was this code removed?

Reply:
@KenitoInc Because the block was repeated
![image](https://user-images.githubusercontent.com/28291332/121400591-70b6be80-c960-11eb-8af4-fdcdeb7f5944.png)